internal/provider/resources/resource_deployment_test.go

```diff
@@ -324,7 +325,228 @@ func TestAcc_ResourceDeploymentStandard(t *testing.T) {
+				Config: astronomerprovider.ProviderConfig(t, true) + standardDeployment(standardDeploymentInput{
+					Name:              scalingSpecDeploymentName,
+					Description:       utils.TestResourceDescription,
+					Region:            "us-east4",
+					CloudProvider:     "GCP",
+					Executor:          "CELERY",
+					SchedulerSize:     "SMALL",
+					IsDevelopmentMode: true,
+					ScalingSpec:       `scaling_spec = {}`,
+				}),
+				ExpectError: regexp.MustCompile(`Inappropriate value for attribute "scaling_spec"`),
+			},
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + standardDeployment(standardDeploymentInput{
+					Name:              scalingSpecDeploymentName,
+					Description:       utils.TestResourceDescription,
+					Region:            "us-east4",
+					CloudProvider:     "GCP",
+					Executor:          "CELERY",
+					SchedulerSize:     "SMALL",
+					IsDevelopmentMode: true,
+					ScalingSpec: `
+						scaling_spec = {
+							hibernation_spec = {}
+						}`}),
+				ExpectError: regexp.MustCompile(`scaling_spec \(hibernation\) must have either override or schedules`),
+			},
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + standardDeployment(standardDeploymentInput{
+					Name:              scalingSpecDeploymentName,
+					Description:       utils.TestResourceDescription,
+					Region:            "us-east4",
+					CloudProvider:     "GCP",
+					Executor:          "CELERY",
+					SchedulerSize:     "SMALL",
+					IsDevelopmentMode: true,
+					ScalingSpec: `
+						scaling_spec = {
+							hibernation_spec = {
+								override = {}
+							}
+						}`}),
+				ExpectError: regexp.MustCompile(`Inappropriate value for attribute "scaling_spec"`),
+			},
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + standardDeployment(standardDeploymentInput{
+					Name:              scalingSpecDeploymentName,
+					Description:       utils.TestResourceDescription,
+					Region:            "us-east4",
+					CloudProvider:     "GCP",
+					Executor:          "CELERY",
+					SchedulerSize:     "SMALL",
+					IsDevelopmentMode: true,
+					ScalingSpec: `
+						scaling_spec = {
+							hibernation_spec = {
+								override = {
+									override_until = "2075-01-01T00:00:00Z"
+								}
+							}
+						}`}),
+				ExpectError: regexp.MustCompile(`Inappropriate value for attribute "scaling_spec"`),
+			},
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + standardDeployment(standardDeploymentInput{
+					Name:              scalingSpecDeploymentName,
+					Description:       utils.TestResourceDescription,
+					Region:            "us-east4",
+					CloudProvider:     "GCP",
+					Executor:          "CELERY",
+					SchedulerSize:     "SMALL",
+					IsDevelopmentMode: true,
+					ScalingSpec: `
+						scaling_spec = {
+							hibernation_spec = {
+								schedules = []
+							}
+						}`}),
+				ExpectError: regexp.MustCompile(`Invalid Attribute Value`), // schedules must have at least one element
```

Comment:
nit - not a blocker, any way we can ensure the assertion on the right attribute here? i.e. `schedules` here